pkg/core/state/notification_event.go

```diff
@@ -12,6 +12,13 @@ import (
 	"github.com/nspcc-dev/neo-go/pkg/vm/vmstate"
 )
 
+const (
+	// mostSignificantBit has the most significant bit of a uint8 set.
+	mostSignificantBit = 0x80
+	// clearMostSignificantBitMask has the most significant bit of a uint8 not set.
+	clearMostSignificantBitMask = mostSignificantBit ^ 0xFF
+)
```

Comment:
The problem is that for this code we don't care if it's most significant bit or any other bit (unless it intersects with some valid VM state). We need to remember why does this kludge exist, hence I'd suggest to use something like `saveInvocationsBit` and `clearSaveInvocationsBitMask` and in the comment describe what is the purpose of this bit/kludge.

Reply:
Like it's said in https://github.com/nspcc-dev/neo-go/pull/3569#discussion_r1926635397.

Reply:
It was not obvious to me what was meant with that discussion as I was just trying to point out there 2 other places using the same magic `0x80` which to me looked identical. I now understand the intent.